include/action.h

```diff
@@ -0,0 +1,56 @@
+#ifndef __LABWC_ACTION_H
+#define __LABWC_ACTION_H
+
+enum action_type {
+	ACTION_TYPE_NONE = 0,
+	ACTION_TYPE_CLOSE,
+	ACTION_TYPE_DEBUG,
+	ACTION_TYPE_EXECUTE,
+	ACTION_TYPE_EXIT,
+	ACTION_TYPE_MOVE_TO_EDGE,
+	ACTION_TYPE_SNAP_TO_EDGE,
+	ACTION_TYPE_NEXT_WINDOW,
+	ACTION_TYPE_PREVIOUS_WINDOW,
+	ACTION_TYPE_RECONFIGURE,
+	ACTION_TYPE_SHOW_MENU,
+	ACTION_TYPE_TOGGLE_MAXIMIZE,
+	ACTION_TYPE_TOGGLE_FULLSCREEN,
+	ACTION_TYPE_TOGGLE_DECORATIONS,
+	ACTION_TYPE_FOCUS,
+	ACTION_TYPE_ICONIFY,
+	ACTION_TYPE_MOVE,
+	ACTION_TYPE_RAISE,
+	ACTION_TYPE_RESIZE,
+};
+
+static const char *ACTION_NAMES[] = {
```

Comment:
It would be better to put this in `action.c` like this:

```
static struct {
    const char *name;
    enum action_type;
} action_names[] = {
    { "None", ACTION_TYPE_NONE },
    { "Close", ACTION_TYPE_CLOSE },
    ...etc
    { NULL, NULL },
};
```

You could then iterate over it in `action_type_from_str()` like so:

```
for (size_t i=0; action_name[i].name; ++i) {
    ...
}
```
This approach will avoid the warning.

Reply:
That is much better. Will do.